### DISCUSSION TITLE
Convert frequency into period property of a time operator with PythonTransform

### DISCUSSION THREAD

**Discussion by frr665** [bonsai-rx/bonsai#1168] [Q&A] [answered]
Dear Bonsai community, 

I'm struggling to modify period property of a timer using a python transform. I want to enter a frequency [Hz] into a float value operator, convert it into a period of time [s] with python transform and use it to set the period property of a timer operator. 
![export_freq](https://user-images.githubusercontent.com/117995542/205114827-1beb522b-c67a-4a84-89c3-5bdc8ddcc113.png)

I was thinking about a function like this one but datetime module don't seems to be available in bonsai. 
```
import datetime
def process(value):
    pulse = 1/freq
    time = datetime.timedelta(seconds=pulse)
    return time
```
Does anyone knows another way to convert a frequency value into a period of time ? 

I hope this make sense and thanks a lot, 
Hugo

**Comment by bruno-f-cruz** [comment, marked as answer]
use an `ExpressionTransform` with [`TimeSpan.FromSeconds(it)`](https://learn.microsoft.com/en-us/dotnet/api/system.timespan.fromseconds?view=net-7.0)

**Reply by frr665** [reply]
Fantastic it worked ! Thanks a lot Bruno. 

For the record this is the working workflow 
![export_freq_2](https://user-images.githubusercontent.com/117995542/205136719-2691fe24-05c5-4c83-81e0-466fed9063c2.png)

Python transform 
```
@returns(float)
def process(value):
  pulse = 1/value
  return pulse
```

Expression transform
```
TimeSpan.FromSeconds(it)
```
